app/assets/javascripts/app/views/notifications_badge_view.js

```diff
@@ -0,0 +1,118 @@
+// @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
+
+app.views.NotificationsBadge = app.views.Base.extend({
+  events:{
+    "click #notifications-badge": "toggleNotifDropdown"
+  },
+
+  initialize: function(){
+    $(document.body).click($.proxy(this.hideNotifDropdown, this));
+
+    this.notifications = [];
+    this.nextPage = 1;
+    this.perPage = 5;
+    this.hasMoreNotifs = true;
+    this.badge = this.$el;
+    this.dropdown = $('#notification_dropdown');
+    this.dropdownNotifications = this.dropdown.find(".notifications");
+    this.ajaxLoader = this.dropdown.find(".ajax_loader");
+  },
+
+  toggleNotifDropdown: function(evt){
+    evt.preventDefault();
+    evt.stopPropagation();
+    if(this.notifDropdownShowing()){ this.hideNotifDropdown(evt); }
+    else{ this.showNotifDropdown(); }
+  },
+
+  notifDropdownShowing: function(){
+    return this.dropdown.css("display") === "block";
+  },
+
+  showNotifDropdown: function(){
+    this.resetParams();
+    this.ajaxLoader.show();
+    this.badge.addClass("active");
+    this.dropdown.css("display", "block");
+    this.dropdownNotifications.addClass("loading");
+    this.getNotifications();
+  },
+
+  hideNotifDropdown: function(evt){
+    var inDropdown = $(evt.target).parents().is(this.dropdown);
+    var inHovercard = $.contains(app.hovercard.el, evt.target);
+    if(!inDropdown && !inHovercard && this.notifDropdownShowing()){
+      this.badge.removeClass("active");
+      this.dropdown.css("display", "none");
+      this.dropdownNotifications.perfectScrollbar('destroy');
+    }
+  },
+
+  notifDropdownScroll: function(){
+    var isLoading = ($('.loading').length === 1);
+    if (this.isBottom() && this.hasMoreNotifs && !isLoading){
+      this.dropdownNotifications.addClass('loading');
+      this.getNotifications();
+    }
+  },
+
+  getParams: function(){
+    if(this.notifications.length === 0){ return{ per_page: 10, page: this.nextPage }; }
+    else{ return{ per_page: this.perPage, page: this.nextPage }; }
+  },
+
+  resetParams: function(){
+    this.notifications.length = 0;
+    this.hasMoreNotifs = true;
+    this.nextPage = 1;
+  },
+
+  isBottom: function(){
+    var bottom = this.dropdownNotifications.prop('scrollHeight') - this.dropdownNotifications.height();
+    var currentPosition = this.dropdownNotifications.scrollTop();
+    return currentPosition + 50 >= bottom;
+  },
+
+  getNotifications: function(){
+    var self = this;
+    $.getJSON(Routes.notifications_path(this.getParams()), function(notifications){
+      $.each(notifications, function(){ self.notifications.push(this); });
+      self.hasMoreNotifs = notifications.length >= self.perPage;
+      self.nextPage++;
```

Comment:
After loading the first notifications the nextPage is `3` because you load 10 notifications at the beginning. You could have an undefined `nextPage` on initialization and here you could set it to `3` if it is undefined and increase it otherwise.